crates/astria-composer/src/searcher/mod.rs

```diff
@@ -76,7 +75,11 @@ impl Searcher {
     /// Errors are returned in the following scenarios:
     /// + failed to connect to the eth RPC server;
     /// + failed to construct a sequencer clinet
-    pub(super) fn from_config(cfg: &Config) -> eyre::Result<Self> {
+    pub(super) fn from_config(
```

Comment:
This constructor is now a misnomer. Please don't add fields to this.

As a side note (not focus of this PR): we should have explicit constructors for all the types instead of treating them like "grab bags" from the config.